lib/src/git.rs

```diff
@@ -1148,13 +1148,32 @@ pub fn reset_head(mut_repo: &mut MutableRepo, wc_commit: &Commit) -> Result<(),
     }
 
     // If there is an ongoing operation (merge, rebase, etc.), we need to clean it
-    // up. This function isn't implemented in `gix`, so we need to use `git2`.
+    // up.
+    //
+    // TODO: Polish and upstream this to `gix`.
     if git_repo.state().is_some() {
-        get_git_backend(mut_repo.store())?
-            .open_git_repo()
-            .map_err(GitExportError::from_git)?
-            .cleanup_state()
-            .map_err(GitExportError::from_git)?;
+        // Based on the files `git2::Repository::cleanup_state` deletes; when
+        // upstreaming this logic should probably become more elaborate to match
+        // `git(1)` behaviour.
+        const STATE_FILE_NAMES: &[&str] = &[
+            "MERGE_HEAD",
+            "MERGE_MODE",
+            "MERGE_MSG",
+            "REVERT_HEAD",
+            "CHERRY_PICK_HEAD",
+            "BISECT_LOG",
+        ];
+        const STATE_DIR_NAMES: &[&str] = &["rebase-merge", "rebase-apply", "sequencer"];
+        let handle = |result: std::io::Result<()>| match result {
+            Err(err) if err.kind() == std::io::ErrorKind::NotFound => Ok(()),
+            _ => result.map_err(GitExportError::from_git),
```

Comment:
nit: better to wrap in `PathError` to provide context

Reply:
Do you think this should be `GitExportError::from_git(PathError { … })` or a new `GitExportError::PathError`? I’ve done the former (since it is a “Git error” and matches how it’ll be reported once this is upstreamed to `gix`) but I can do the latter if preferred.

Reply:
I also mean the former. Thanks.